docs/create-packages/Explicit-assembly-reference.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Explicit Assembly References
+description: Make a subset of assemblies in the package available to the compiler, while all assemblies are available at runtime.
+author: zivkan
+ms.author: zivkan
+ms.date: 05/24/2019
+ms.topic: conceptual
+---
+
+# Explicit Assembly References
+
+Explicit assembly references allows a subset of assemblies to be used for intellisense and compiling, while all assemblies are available at run-time. `PackageReference` and `packages.config` work differently, and as a result package authors need to take care to create the package to be compatible with both project types.
+
+> [!Note]
+> Explicit assembly references are related to .NET assemblies. It is not a method to distribute native assemblies that are P/Invoked by a managed assembly.
+
+## `PackageReference` support
+
+When a project uses a package with `PackageReference` and the package contains a `ref\<tfm>\` directory, NuGet will advertise those assembles as compile-time assets, while the `lib\<tfm>\` assemblies are advertised as runtime assets. Assemblies in `ref\<tfm>\` are not used at runtime. This means it is necessary for any assembly in `ref\<tfm>\` to have a matching assembly in either `lib\<tfm>\` or a relevant `runtime\` directory, otherwise runtime errors will likely occur. Since assemblies in `ref\<tfm>\` are not used at runtime, they may be [metadata-only assemblies](https://github.com/dotnet/roslyn/blob/master/docs/features/refout.md) to reduce package size.
+
+> [!Important]
+> If a package contains the nuspec `<references>` element (used by `packages.config`, see below) and does not contain assemblies in `ref\<tfm>\`, NuGet will advertise the assemblies listed in the nuspec `<references>` element as both the compile and runtime assets. This means there will be runtime exceptions when the referenced assemblies need to load any other assembly in the `lib\<tfm>\` directory.
+
+> [!Note]
+> If the package contains a `runtime\` directory, NuGet may not use the assets in the `lib\` directory.
```

Comment:
What do we mean by "NuGet my not use..."? I am guessing it does not mean at runtime since we started the topic we state "while all assemblies are available at run-time"
Also, "may not" does not inspire confidence.

Reply:
Partly it's because I don't know the answer, partly I think there are cases when NuGet could use the `lib/` assets.

Consider a package that has the files `lib/netcoreapp2.2/pkg.dll`, `lib/net472/pkg.dll`, `ref/netcoreapp2.2/pkg.dll`, `ref/net472/pkg.dll` and `runtimes/win-x64/lib/netcoreapp2.2/pkg.dll`. So, the package has `netcoreapp2.2` and `net472` `lib` and `ref` files, and it has runtime file only for `netcoreapp2.2` on `win-x64`. What happens if the project using the package target `net472`? I expect it to use the `lib\` assets. But if the project targets `netcoreapp2.2` and is running on `win-x64`, it should use the `runtimes\` assets. But what happens if the project is running on `linux` or `win-x86`? My guess is it uses the `lib\` assets.